CMakeLists.txt

```diff
@@ -1,5 +1,5 @@
+cmake_minimum_required (VERSION 3.0)
```

Comment:
@ignatenkobrain @cgwalters I do not think that this PR is a good idea, as the feature being used was introduced in CMake 3.0.0. CMake 2.8.12 doesn't have these variables defined.